### ISSUE TITLE
[Accessibility] larger text option?

### ISSUE THREAD

**Issue by bleaky** [tootsuite/mastodon#873]
simple enough. the normal size text is hard to read for me, and i was lucky enough to be able to know how to use my browser's tools to enlarge them myself but it'd be nice and accessible if there were a font size option.

* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by kennethlarsen** [comment]
I also struggle a bit with this. I would suggest adding it as an option in the settings like this: 
![screenshot_20170404_214043](https://cloud.githubusercontent.com/assets/1408595/24675495/a52e2cd8-197f-11e7-87ee-e1a0be4791a0.png)

**Comment by Maverynthia** [comment]
Yes this. I mostly browse Mastodon while relaxing in my chair. I have one userscript and the zoom in feature, but sometimes it makes things act funny.

**Comment by ghostfromicosahedron** [comment]
Yes please!! Yes to all accessibility requests! Right now, if I zoom in on my browser, it becomes readable but then I've got to scroll left-right to see all the columns. If we could make the text bigger in our settings (which would show fewer toots per column), we could still see all the columns on one screen, but at a magnification that works for tired or disabled eyes and take that setting preference with us to other devices.

**Comment by lanodan** [comment]
The real good thing would be to use the browser preferences, you can set your browser to use larger/smaller font (than the common default of 16px) and websites should respect that.

**Comment by jpdevries** [comment]
I agree with @lanodan that browser/OS preferences should be respected. The issue is that Mastodon currently sets type size in pixels. In the below GIF example, notice that when I change my font size preference in Chrome from Medium to Very Large none of the type scales :(

Therefore, I think the best solution would be to set type in relative units regardless of offering any font size preferences within Mastodon.

<details>
  <summary>GIF</summary>
  <img src="http://j4p.us/1x241v233X1j/mastodon_fontsize.gif" alt="Animated GIF showing how Mastodon does not respect Chrome font size settings" />
</details>
<br>

ps: I would totally 👍 an accessibility label for organizing issues.  
Such a label could make making Mastodon accessible more accessible 🙃

**Comment by navjotjsingh** [comment]
Any updates?

**Comment by ysulyma** [comment]
just joined Mastodon, finding it hard to read :(

**Comment by lehni** [comment]
I agree, the type size seems really quite small compared to almost any other website nowadays.

**Comment by Freeplayg** [comment]
Would need to replace px with em, so it's easier to just change the font size in the body element to have it reflect through the whole ui https://developer.mozilla.org/en-US/docs/Web/CSS/font-size#ems

**Comment by regulardesigner** [comment]
Yes, it's a good idea to manage font size change that way @Freeplayg but, I guess the code changes needed to make it will be a huge amount of work.

**Comment by brennanyoung** [comment]
Text zoom is one of those a11y features that gets most use "in the wild", even from people who don't identify as disabled. This should certainly be prioritised fairly high. If there is a legacy of using px, then even more reason to attend to it sooner rather than later.

Rule of thumb: Reserve px for hairline borders and media queries, or for specifying the root font size.

**Comment by mgifford** [comment]
I like this implementation that covers both the font size and spacing https://docs.fluidproject.org/infusion/development/preferenceseditor

**Comment by regulardesigner** [comment]
> The real good thing would be to use the browser preferences, you can set your browser to use larger/smaller font (than the common default of 16px) and websites should respect that.

This could be the first thing to do before creating a zoom fonts option.

**Comment by brennanyoung** [comment]
Agree, leaning on the browser's built-in text zoom is optimal. The keyboard shortcut (ctrl+ or cmd+) is well established and reasonably predictable. No need to reinvent the wheel. 

Even native mobile can (and should) hook into the (adjustable) system font size, but I guess that's another repo.

Text spacing adjustment does not have a well-established keyboard shortcut (yet), but an implementation activated by a keyboard shortcut would be a fair MVP, allowing the development of a GUI preference/setting to be postponed while the effects on layout are ironed out.

(For any keyboard shortcut, be mindful of [WCAG SC 2.1.4](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts) though, TL;DR: always include a modifier key in the shortcut.).

**Comment by venteto** [comment]
Yes, please make this a preference in settings, in a way that won't totally kill the "simple" 3-column layout (I won't even try the Tweetdeck-esque layout, particiuarly when I keep my laptop browser's sidebar open permanently for tabs, bookmarks and such).

As it stands, in the non-advanced layout, using the Firefox ability to simply "zoom" the page in Mastodon starts hiding the lefthand compose post column first, and then eventually the righthand navigation column as well.

I suspect if I could just go into settings, and set my default font size just for reading and composing posts to, say, 22px (as well as, perhaps, specifying padding in areas like the post interaction toolbar), I might still be able to retain all the columns even while having my Firefox sidebar open (which leaves me typically about 900 horizontal pixels left for the viewport). I don't necessarily need to be able to change the font size for other parts of the web UI, but maybe these could also be a separate, optional preference for those who might desire to tweak different parts of the UI independently.
